code/game/machinery/vending/vending.dm

```diff
@@ -168,9 +170,20 @@
 			attack_hand(user)
 		return
 	else if(istype(W, /obj/item/coin))
-		if(user.drop_inv_item_to_loc(W, src))
-			coin = W
-			to_chat(user, "\blue You insert the [W] into the [src]")
+		var/obj/item/coin/C = W
+		if(coin)
+			to_chat(user, "<span class='warning'>[src] already has [coin] inserted</span>")
+			return
+		if(!premium.len && !istype(src,/obj/machinery/vending/shared_vending))
+			to_chat(user, "<span class='warning'>[src] doesn't have a coin slot.</span>")
+			return
+		if(C.flags_token & tokensupport || C.flags_token & TOKEN_ALL)
```

Comment:
There is no need for this second check if the TOKEN_ALL actually has all bits on as recommended before.